File: app/src/androidTest/java/com/example/benchmarks/models/TestBenchmarksModule.java
package com.example.benchmarks.models;

import com.example.benchmarks.models.benchmark.Benchmark;
import com.example.benchmarks.models.benchmark.TestOperationsCollections;
import com.example.benchmarks.models.benchmark.TestOperationsMaps;

import javax.inject.Named;

import dagger.Module;
import dagger.Provides;

@Module
public class TestBenchmarksModule extends BenchmarksModule {

    @Provides
    @Named("collections")
    public Benchmark provideOperationsCollections() {
        return new TestOperationsCollections();
    }

    @Provides
    @Named("maps")
    public Benchmark provideOperationMaps() {
        return new TestOperationsMaps();
    }
}

File: app/src/androidTest/java/com/example/benchmarks/models/TestBenchmarksModule.kt
package com.example.benchmarks.models

import com.example.benchmarks.models.benchmark.Benchmark
import com.example.benchmarks.models.benchmark.TestOperationsCollections
import com.example.benchmarks.models.benchmark.TestOperationsMaps
import dagger.Module
import dagger.Provides
import javax.inject.Named

@Module
class TestBenchmarksModule : BenchmarksModule() {
    @Provides
    @Named("collections")
    override fun provideOperationsCollections(): Benchmark {
        return TestOperationsCollections()
    }

    @Provides
    @Named("maps")
    override fun provideOperationMaps(): Benchmark {
        return TestOperationsMaps()
    }
}